src/lib/change-email-helpers.js

```diff
@@ -0,0 +1,92 @@
+const helpers = require('../lib/helpers');
+const uuid = require('uuid/v4');
+const moment = require('moment');
+const { pool } = require('../lib/connectors/db');
+const { EmailChangeError, checkEmailAddressInIDM } = require('../controllers/change-email');
+const Repository = require('@envage/hapi-pg-rest-api/src/repository');
+
+const usersRepo = new Repository({
+  connection: pool,
+  table: 'idm.users',
+  primaryKey: 'user_id'
+});
+
+const emailChangeRepo = new Repository({
+  connection: pool,
+  table: 'idm.email_change_verification',
+  primaryKey: 'email_change_verification_id'
+});
+
+const createEmailChangeRecord = async (userId, authenticated) => {
+  const data = {
+    email_change_verification_id: uuid(),
+    user_id: userId,
+    new_email_address: null,
+    authenticated,
+    verification_code: null,
+    date_created: moment().format('YYYY-MM-DD HH:mm:ss'),
+    date_verified: null
+  };
+  return emailChangeRepo.create(data);
+};
+
+const getRecordsByUserId = async userId => {
+  return emailChangeRepo.find({
+    user_id: userId,
+    date_created: { $gte: moment().subtract(1, 'days').toISOString() }
+  });
+};
+
+const authenticateUserById = async (userId, password) => {
+  const user = usersRepo.find({ user_id: userId });
+  if (user) {
+    return helpers.compareHash(password, user.password);
+  }
+  return 404;
```

Comment:
I'm not sure we should be seeing HTTP status codes in the data access layer.
We should probably return just the data and let the controller decide what HTTP codes to generate.